force-app/main/default/lwc/candidateRelated/candidateRelated.js

```diff
@@ -0,0 +1,132 @@
+import { LightningElement, wire, track } from "lwc";
+import { CurrentPageReference } from "lightning/navigation";
+import selectCandidatesRelatedToPosition from "@salesforce/apex/CandidateRelatedController.selectCandidatesRelatedToPosition";
+import selectCountOfRelatedCandidates from "@salesforce/apex/CandidateRelatedController.selectCountOfRelatedCandidates";
+import Related_Candidates from "@salesforce/label/c.Related_Candidates";
+import Empty_List from "@salesforce/label/c.Empty_List";
+import selectJobApplicationRelatedToCandidate from "@salesforce/apex/CandidateRelatedController.selectJobApplicationRelatedToCandidate";
+import selectCandidateById from "@salesforce/apex/CandidateRelatedController.selectCandidateById";
+import getCustomMetadataDefaultForRelatedCandidate from "@salesforce/apex/CandidateRelatedController.getCustomMetadataDefaultForRelatedCandidate";
+import CandidateAndJobApplicationModal from "c/candidateAndJobApplicationModal";
+import { ShowToastEvent } from "lightning/platformShowToastEvent";
+
+export default class CandidateRelated extends LightningElement {
+  labels = { Related_Candidates, Empty_List };
+  @wire(CurrentPageReference)
+  currentPageReference;
+  recordId;
+  currentPage = 1;
+  @track candidates = [];
+  pageSize;
+  @track candidateCount;
+  isAvatarsShowed;
+
+  connectedCallback() {
+    this.recordId = this.currentPageReference.attributes.recordId;
+  }
+
+  @wire(getCustomMetadataDefaultForRelatedCandidate)
+  Metadatarecord({ data, error }) {
+    if (data) {
+      this.pageSize = data.Candidates_Per_Page__c;
+      this.isAvatarsShowed = data.Is_Avatars_Shown__c;
+    } else if (error) {
+      this.showErrorToast(this.generateErrorMessage(error));
+    }
+  }
+
+  @wire(selectCountOfRelatedCandidates, { positionId: "$recordId" })
+  CandidateCount({ data, error }) {
+    if (data) {
+      this.candidateCount = data[0].expr0;
+    } else if (error) {
+      this.showErrorToast(this.generateErrorMessage(error));
+    }
+  }
+
+  @wire(selectCandidatesRelatedToPosition, {
+    positionId: "$recordId",
+    pageOffset: "$pageOffset",
+    pageSize: "$pageSize"
+  })
+  Candidates({ data, error }) {
+    if (data) {
+      this.candidates = data.map((record) => {
+        return {
```

Comment:
You can leave it as it is, but usually we use DTO pattern to cover this task. (FYI)

2 problems that it solves:
1) You don't need to do it on UI controller
2) You return only the required information in the correct form.